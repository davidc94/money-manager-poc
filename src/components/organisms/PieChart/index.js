import React, { Component } from 'react';
import {
  AppRegistry, StyleSheet, ScrollView, StatusBar, Text, View,
} from 'react-native';
import PieChart from 'react-native-pie-chart';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  title: {
    fontSize: 24,
    margin: 5,
  },
});

export default class test extends Component {
  render() {
    const chart_wh = 200;
    const series = [0.5, 1.75, 2, 3.55, 5.67];
    const sliceColor = ['#F44336', '#2196F3', '#FFEB3B', '#4CAF50', '#FF9800'];

    return (
      <ScrollView style={{ flex: 1 }}>
        <View style={styles.container}>
          <StatusBar
            hidden
          />
          <Text style={styles.title}>Doughnut chart</Text>
          <PieChart
            chart_wh={chart_wh}
            series={series}
            sliceColor={sliceColor}
            doughnut
            coverRadius={0.45}
            coverFill="#FFF"
          />
        </View>
      </ScrollView>
    );
  }
}

AppRegistry.registerComponent('test', () => test);
