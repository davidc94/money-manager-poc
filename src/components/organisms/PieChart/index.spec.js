import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PieChart from './index';

configure({ adapter: new Adapter() });

describe('pie chart', () => {
  const render = shallow(<PieChart />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
