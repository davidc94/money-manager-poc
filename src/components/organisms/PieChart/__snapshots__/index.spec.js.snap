// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pie chart renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <test />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "chart_wh": 200,
      "coverFill": "#eee",
      "coverRadius": 0.65,
      "doughnut": true,
      "series": undefined,
      "sliceColor": Array [
        "#F44336",
        "#2196F3",
        "#FFEB3B",
        "#4CAF50",
        "#FF9800",
      ],
      "style": Object {
        "alignSelf": "center",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "chart_wh": 200,
        "coverFill": "#eee",
        "coverRadius": 0.65,
        "doughnut": true,
        "series": undefined,
        "sliceColor": Array [
          "#F44336",
          "#2196F3",
          "#FFEB3B",
          "#4CAF50",
          "#FF9800",
        ],
        "style": Object {
          "alignSelf": "center",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
